library/model/book.py

```diff
@@ -101,7 +108,31 @@ def __eq__(self, other):
         return NotImplemented
 
     def __str__(self):
-        return BookSerializer().serialize(self, "JSON")
+        return self.serialize(self, "JSON")
+
+    def serialize(self, book_format: str):
+        if book_format == "JSON":
+            book_info = {
+                "id": self.isbn,
+                "title": self.title,
+                "authors": [author.get_fullname() for author in self.authors],
+                "available_items": self.existing_items - self.borrowed_items,
+                "borrowed_items": self.borrowed_items,
+            }
+            return json.dumps(book_info)
+        elif book_format == "XML":
+            book_info = et.Element("book", attrib={"id": self.isbn})
+            title = et.SubElement(book_info, "title")
+            title.text = self.title
+            authors = et.SubElement(book_info, "authors")
+            authors.text = ", ".join([author.get_fullname() for author in self.authors])
+            avail = et.SubElement(book_info, "available")
+            avail.text = str(self.existing_items - self.borrowed_items)
+            authors = et.SubElement(book_info, "borrowed")
+            authors.text = str(self.borrowed_items)
+            return et.tostring(book_info, encoding="Unicode")
+        else:
+            raise ValueError(format)
 
 
 class BorrowedBook(Book):
```

Comment:
The BorrowedBook does not need to be inherited from the Book. It could also have the Book instance